### PR TITLE
OPENEUROPA-3022: Drupal 8.8 upgrade & permission to view any unpublished media

### PR DESCRIPTION
**Pull request by upchuk**
* Upgrade to Drupal 8.8 (but allow also 8.7)
* Create permission and access handling for viewing any unpublished media entities

### CODE REVIEW

File: tests/Kernel/MediaAccessTest.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    $this->assertTrue($media->access('view', $this->currentUser));
+    $this->assertCount(1, $view->result);
+
+    $media->set('status', 0);
+    $media->save();
+
+    // Reset the static cache only.
+    $access_handler->resetCache();
+
+    // The current user should not access to the published media.
```
Comment (nagyad) [inline tests/Kernel/MediaAccessTest.php]: ... should not have access to the unpublished media.

File: tests/Kernel/MediaAccessTest.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    // Give the user the role to view any unpublished media.
+    /** @var \Drupal\user\RoleInterface $role */
+    $role = Role::load($this->currentUser->getRoles(TRUE)[0]);
+    $role->grantPermission('view any unpublished media');
+    $role->save();
+
+    $access_handler->resetCache();
+    // The user should have access to the media based on the permission.
```
Comment (nagyad) [inline tests/Kernel/MediaAccessTest.php]: The user should have access to the unpublished media based on the permission.

**Comment by upchuk** [comment]
> Why do we have to do this on our side? Isn't core media should provide such a permission?

It doesn't provide this permission and the one it provides (administer media) is too much for what we want to give. So a generic `view any unpublished media` could be created for this reason.

File: tests/Kernel/MediaAccessTest.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+  /**
+   * Executes a View.
+   *
+   * @param \Drupal\views\ViewExecutable $view
+   *   The view to execute.
+   * @param string $display_id
+   *   The display id.
+   *
+   * @internal param string $display The display id*   The display id
```
Comment (imanoleguskiza) [inline tests/Kernel/MediaAccessTest.php]: I am a bit confused about this documentation. What does it mean?

Comment (upchuk) [reply tests/Kernel/MediaAccessTest.php]: Nothing, it was a mistake. Fixed. Thanks

**Pull request merged**